### PR TITLE
Fix pav plot

### PR DESCRIPTION
**Pull request by wowtor**
added docs, removed redundant code; this fixes a bug in pav plot

### CODE REVIEW

File: lir/calibration.py [hunk 1]
```diff
@@ -194,32 +191,19 @@ def __init__(self, add_one=False):
         self._ir = IsotonicRegression()
 
     def fit(self, X, y, **fit_params):
```
Comment (rolfypma) [inline lir/calibration.py:221]: mooier!

File: lir/calibration.py [hunk 2]
```diff
@@ -194,32 +191,19 @@ def __init__(self, add_one=False):
...
 
         return self
 
     def transform(self, X):
-        if isinstance(X, np.matrix):
-            X = X.A1
-
-        posterior = self._ir.transform(X)
-
-        self.p0 = (1 - posterior)
```
Comment (rolfypma) [inline lir/calibration.py:219]: er zijn functies (iig plot functies, misschien meer) die ervan uit gaan dat de modellen p0 en p1 invullen. Is dit een feature die we willen behouden of willen we dit eruit schrijven? Voor nu gaat deze verandering functies breken.

File: lir/metrics.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    with np.errstate(divide='ignore'):
+        lrs0, lrs1 = Xy_to_Xn(lrs, y)
+        cllr0 = np.mean(np.log2(1 + lrs0))
+        cllr1 = np.mean(np.log2(1 + 1/lrs1))
+        return .5 * (cllr0 + cllr1)
+
+
+def cllr_min(lrs, y):
+    """
+    Calculates the discriminative power for a series of likelihood ratios.
```
Comment (rolfypma) [inline lir/metrics.py]: Netter:
Estimates the discriminative power from a collection of likelihood ratios

File: lir/metrics.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    -------
+    cllr_min
+        the log likelihood ratio cost
+    """
+    cal = IsotonicCalibrator()
+    lrmin = cal.fit_transform(to_probability(lrs), y)
+    return cllr(lrmin, y)
+
+
+def calculate_cllr(lr_class0, lr_class1):
```
Comment (rolfypma) [inline lir/metrics.py]: verwarrend dat zowel deze als cllr() bestaat met gelijke docstrings. de andere intern (_cllr) maken, en docstrings aanpassen?

File: lir/plotting.py [hunk 5]
```diff
@@ -306,3 +308,55 @@ def makeplot_accuracy(scorer, density_function, X0_train, X1_train, X0_calibrate
...
+
+
+def plot_pav(lrs, y, show_scatter=True, savefig=None, show=None, kw_figure={}):
+    """
+    Generates a plot of pre- versus post-calibrated LRs using Pool Adjacent Violators (PAV).
+
+    Parameters
+    ----------
+    lrs : numpy array of floats
+        Pre-calibrated likelihood ratios
```
Comment (rolfypma) [inline lir/plotting.py]: precalibrated is ambigu. mssn 'likelihood ratios before the PAV transform' ?

File: lir/plotting.py [hunk 6]
```diff
@@ -306,3 +308,55 @@ def makeplot_accuracy(scorer, density_function, X0_train, X1_train, X0_calibrate
...
+    y : numpy array
+        Labels corresponding to lrs (0 for Hd and 1 for Hp)
```
Comment (rolfypma) [inline lir/plotting.py:322]: is dit een conventie de we het hele package aan gaan houden? Dat hadden we nog niet.

File: lir/plotting.py [hunk 7]
```diff
@@ -306,3 +308,55 @@ def makeplot_accuracy(scorer, density_function, X0_train, X1_train, X0_calibrate
...
+        If True, show individual LRs
+    on_screen : boolean
+        If True, show the plot on screen
+    path : str
+        If not None, write the figure to a file
+    kw_figure : dict
+        Keyword arguments that are passed to matplotlib.pyplot.figure()
+    """
+    pav = IsotonicCalibrator()
+    pav_lrs = pav.fit_transform(util.to_probability(lrs), y)
```
Comment (rolfypma) [inline lir/plotting.py]: is de to_probability nodig? Zou voor pav niet uit moeten maken?

**Review by rolfypma — APPROVED** [review]
er staat nog een ref naar de oude pav.plot in lr.py in de root folder. Deze mag nog weg/aangepast.

File: lir/plotting.py (hunk 6 above)
Comment (wowtor) [reply lir/plotting.py:322]: Een conventie hadden we inderdaad nog niet echt. Deze stijl werd elders wel gebruikt. Ik heb geen grote voorkeur voor een stijl maar een conventie is zeker een goed idee. Hebben we eigenlijk als team een conventie?

File: lir/metrics.py (hunk 4 above)
Comment (wowtor) [reply lir/metrics.py]: Klopt. Maar eigenlijk is het juist deze die niet echt de lading dekt, want hij berekent veel meer statistieken dan de cllr.

File: lir/plotting.py (hunk 7 above)
Comment (wowtor) [reply lir/plotting.py]: klopt, aangepast

File: lir/plotting.py (hunk 5 above)
Comment (wowtor) [reply lir/plotting.py]: ok

File: lir/metrics.py (hunk 3 above)
Comment (wowtor) [reply lir/metrics.py]: ok

File: lir/calibration.py (hunk 2 above)
Comment (wowtor) [reply lir/calibration.py:219]: ok, terug gezet

File: lir/plotting.py (hunk 6 above)
Comment (rolfypma) [reply lir/plotting.py:322]: impliciet. lijkt mij een goed idee overal in lir over te stappen van x0, x1 op x1, x2 en labels 1 en 2 te gebruiken voor y. Veel moeilijker om fouten te maken.

Comment (wowtor) [reply lir/plotting.py:322]: Ah je bedoelt de conventie voor y labels. Ik dacht docstrings :)

Voor y labels is door de hele package heen dit dacht ik wel de conventie, niet? ik vind labels 1 en 2 best verwarrend. Ik neem aan dat je met 1 en 2 bedoelt dat ze overeen komen meet h1 en h2. Dan corresondeert het hogere label (2) met de lagere score (0) en dat vind ik niet logisch. We zitten min of meer vast aan de scoring 0..1 omdat dat de standaard is in o.a. sklearn.

File: lir/metrics.py (hunk 4 above)
Comment (TimoMatzen) [reply lir/metrics.py]: Hier liep ik zelf ook al tegenaan deze week, vond het inderdaad niet intuitief dat calculate_cllr zoveel meer teruggeeft dan alleen de cllr.
